### PR TITLE
feat(discover-tags): Add Show More button

### PR DESCRIPTION
**Pull request by narsaynorath**
Adds a "Show More" button to the tag summary component in Discover. Users will be able to fetch tags in pages of 10.

The following is a screenshot of the new UI with our demo org.

Screenshot 2023-06-26 at 10 59 02 AM

### CODE REVIEW

File: static/app/views/discover/tags.tsx [hunk 1]
```diff
@@ -152,9 +180,29 @@ class Tags extends Component<Props, State> {
...
-          {tags.map((tag, index) => this.renderTag(tag, index))}
-        </TagFacetsList>
+        <Fragment>
+          <StyledTagFacetList>
+            {tags.map((tag, index) => this.renderTag(tag, index))}
+          </StyledTagFacetList>
+          {hasMore &&
+            (loading ? (
+              this.renderPlaceholders()
+            ) : (
```
Comment (k-fish) [inline static/app/views/discover/tags.tsx:190]: Does `Show More` show if there is an error? 🤔

**Review by k-fish — APPROVED** [review]
Code looks fine, in terms of api response we may consider not sorting alphabetically now that we show multiple pages (since the sort doesn't span pages).

File: static/app/views/discover/tags.tsx (hunk 1 above)
Comment (narsaynorath) [reply static/app/views/discover/tags.tsx:190]: No, if there's an error then there's a conditional above that renders an error state for the whole sidebar

**Comment by narsaynorath** [comment]
> in terms of api response we may consider not sorting alphabetically now that we show multiple pages

I'll merge this as-is for now, but I'll follow up on the sort order. I think that if there's just a single page, it makes sense to sort it, but if there aren't then maybe we shouldn't sort it. Although, not having a sort order could make it more confusing. We'll see!

**Comment by k-fish** [comment]
@narsaynorath sounds good 👍

**Pull request merged**